docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -749,103 +788,161 @@ this introduces problematic scenarios that can happen during an upgrade:
+  - `node2` handles the packet exactly as with the old topology
+
+For comparison, fully upgraded scenario for ingress service via node1 to local pod1 `203.203.0.2` looks like this:
+- incoming packet gets DNAT'ed to the `serviceIP` and SNAT'ed to the node masq IP (src=`169.254.0.2`), then comes to the GR
+- GR does its usual DNAT to the backend pod IP (dst=`203.203.0.2`) and SNAT to the joinIP (src=`100.65.0.2`)
+- GR sends the packet via `rtotr-GR-node1` to the `transit_router`
+- `transit_router` sends the packet (dst=`203.203.0.2`) directly to the `layer2-switch`, done
+Now reply:
+- pod1 (`203.203.0.2`) replies to `100.65.0.2` -> default GW (`203.203.0.1`)
+- `transit_router` will ARP for `100.65.0.2` and get a reply from `GR-node1` via port `rtotr-GR-node1` with `0a:58:64:41:00:02`
+- `transit_router` will do route lookup and choose between `100.65.0.0/16` as a connected route from `100.65.255.254`
+  via port `trtos-layer2-switch` and `100.65.0.2/32` via `100.88.0.6` static route
+- `transit_router` will choose `100.65.0.2/32` with the longest-prefix match and send it to the `GR-node1`
+- the usual un-DNAT (dst=`169.254.0.2`), un-SNAT (src=`serviceIP`) happens here before being sent out
+
+Some of the scenarios that we care about:
+- Ingress traffic coming to GR via service
+  - upgraded GR -> non-upgraded node pod: we SNAT to the same joinIP as before, reply traffic comes back to the 
+    `transit_router` because `trtos-layer2-switch-upgrade` port is configured with the GW MAC.
+  - non-upgraded GR -> upgraded node pod: uses the same joinIP as before. Reply uses the newly assigned dummyIP
+    from the join subnet, to enable routing from the upgraded node pod to the joinIP subnet. Without the dummy IP
+    transit_router doesn't know what to do with the `dst: 100.65.0.3` traffic.
+- Egress IP traffic: uses the same joinIP to route as before, works similar to service traffic.
+
+The same GR MAC address is used on the GR port (`rtotr-GR-node1`) and the transit router port (`trtos-layer2-switch-upgrade`).
+That's fine because they're different L2 domains.
+
+Dummy IP can be resued on all nodes, since it is never used as a source or destination IP, it is only used for the routing decision.
+
+joinIP is not really needed in the new topology, but we will have to keep it during the upgrade, and we can't
+remove it after the upgrade is done, because there will be conntrack entries using it. Not too big of a deal,
+but a side effect of supporting the upgrade.
+There is a way to remove joinIP eventually, which requires 2 more upgrades:
+1. On the next upgrade replace all NATs using joinIPs with the transitIPs (at this point the new topology is used and all nodes understand
+transit IPs). This can only be done in the same manner as this upgrade (i.e. after the node reboots) to avoid disruption.
+2. On the next-after-next upgrade we can safely remove joinIPs from the topology, since they shouldn't be used anymore.
```

Comment:
I wouldn't put this here, it just adds confusion that makes it look this is part of this enhancement. I would just say that joinIP deprecation and removal will not be part of this enhancement and may be tackled in a future release.